tests/bdd/features/volume/topology/pool-affinity.feature

```diff
@@ -0,0 +1,30 @@
+Feature: Volume Topology
+  The volume topology feature is for selection of particular pools to schedule a replica on.
+  All the volume topology key and values should have an exact match with pool labels, although pools
+  can have extra labels, for the selection of that pool. No volume topology symbolizes the volume replicas
+  can be scheduled on any available pool.
+
+  Background:
+    Given a control plane, three Io-Engine instances, three pools
+
+  Scenario: suitable pools which contain volume topology labels
+    Given a request for a volume with its topology inclusion labels matching certain pools with same labels
```

Comment:
I wonder if these might be too generic, maybe better with actual examples, and you can use Scenario Outline and Examples table for both input and outcome, wdyt @Abhinandan-Purkait ?

Reply:
IMHO, there are 4 test cases only , This PR can be taken as is . Based on time i can try out the  Scenario Outline and Examples